lib/Migration/Version23000Date2022022912000001.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * @copyright Copyright (c) 2021 Carl Schwan <carl@carlschwan.eu>
+ *
+ * @author Carl Schwan <carl@carlschwan.eu>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+declare(strict_types=1);
+
+namespace OCA\MonthlyStatusEmail\Migration;
+
+use Closure;
+use OCP\DB\ISchemaWrapper;
+use OCP\Migration\SimpleMigrationStep;
+use OCP\Migration\IOutput;
+use OCP\IDBConnection;
+
+class Version23000Date2022022912000001 extends SimpleMigrationStep {
+
+	/** @var IDBConnection */
+	protected $connection;
+
+	public function __construct(IDBConnection $connection) {
+		$this->connection = $connection;
+	}
+
+	public function preSchemaChange(IOutput $output, Closure $schemaClosure, array $options) {
+		$query = $this->connection->getQueryBuilder();
+
+		// delete all but one entries for each user_id
+		$query->select($query->func()->max('id', 'newest_entry'), 'user_id')
+			->from('notification_tracker')
+			->groupBy('user_id');
+
+		$result = $query->executeQuery();
+		$rows = $result->fetchAll();
+		foreach ($rows as $row)
+			$delete = $this->connection->getQueryBuilder();
```

Comment:
do this outside of the loop and use parameters:
https://github.com/nextcloud/spreed/blob/647086cae20f4061a2b8ecfb5a7e5c7e511905b1/lib/Migration/Version11000Date20200922161218.php#L100-L118